File: test/benchmark/fib.go
package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	a := 0
	b := 1

	for i := 0; i < n; i++ {
		a, b = b, a+b
	}
	return a
}

func main() {
	start := time.Now()

	answer := fib(50)

	elapsed := time.Since(start)

	fmt.Printf("Found answer %d\nElapsed: ", answer)
	fmt.Println(elapsed.Seconds())
}

File: test/benchmark/fib.rs
use std::time::Instant;

pub fn fib(n: i32) -> u64 {
	if n < 0 {
		panic!("{} is negative!", n);
	} else if n == 0 {
		panic!("zero is not a right argument to fibonacci()!");
	} else if n == 1 {
		return 1;
	}

	let mut sum = 0;
	let mut last = 0;
	let mut curr = 1;
	for _i in 1..n {
		sum = last + curr;
		last = curr;
		curr = sum;
	}
	sum
}

fn main() {
    let start = Instant::now();
    let answer = fib(50);
    let elapsed = start.elapsed().as_secs();

    println!("Found answer {}", answer);
    println!("Elapsed: {}", elapsed);
}
